apps/src/templates/teacherDashboard/SectionsSharingButton.jsx

```diff
@@ -0,0 +1,76 @@
+import React, {Component, PropTypes} from 'react';
+import {connect} from 'react-redux';
+import Button from '../Button';
+import ChangeShareSettingDialog from './ChangeShareSettingDialog';
+import {sectionShape} from './shapes';
+import i18n from "@cdo/locale";
+
+class SectionsSharingButton extends Component {
+  static propTypes = {
+    sectionId: PropTypes.number.isRequired,
+    // Provided by Redux
+    section: sectionShape,
+    students: PropTypes.array
+  };
+
+  state = {
+    isDialogOpen: false,
+  };
+
+  openDialog = () => this.setState({isDialogOpen: true});
+  closeDialog = () => this.setState({isDialogOpen: false});
+
+  render() {
+    const {section, students} = this.props;
+    if (!section || !students) {
+      // Render nothing for not-yet-loaded section information
+      return null;
+    }
+
+    // Show the enable button if all students have sharing disabled
+    // Otherwise show the disable button
+    let showEnabledButton = true;
+    for (let i = 0; i < students.length; i++){
+      showEnabledButton = showEnabledButton && students[i].sharingDisabled;
+    }
```

Comment:
Could also write this as `const showEnabledButton = !students.some(student => !student.sharingDisabled)` (though you'd need to double check whether I got the conditionals right.